kube-client/src/config/mod.rs

```diff
@@ -271,6 +273,45 @@ impl Config {
         })
     }
 
+    /// Override configuration based on environment variables
+    ///
+    /// This is only intended for use as a debugging aid, and the specific variables and their behaviour
+    /// should **not** be considered stable across releases.
+    ///
+    /// Currently, the following overrides are supported:
+    ///
+    /// - `KUBE_RS_DEBUG_IMPERSONATE_USER`: A Kubernetes user to impersonate, for example: `system:serviceaccount:default:foo` will impersonate the `ServiceAccount` `foo` in the `Namespace` `default`
+    /// - `KUBE_RS_DEBUG_IMPERSONATE_GROUP`: A Kubernetes group to impersonate, multiple groups may be specified by separating them with commas
+    /// - `KUBE_RS_DEBUG_OVERRIDE_URL`: A Kubernetes cluster URL to use rather than the one specified in the config, useful for proxying traffic through `kubectl proxy`
+    #[tracing::instrument(level = "warn")]
+    pub fn apply_debug_overrides(&mut self) {
```

Comment:
How about taking the `DEBUG` parts out of these names, e.g.:

```suggestion
    pub fn allow_env_overrides(&mut self) {
```

`allow_debug_overrides` sounds a bit scary - like something you should not do in production - but yet we are doing it in the main recommended entry `Config::infer` - which will cause people to think twice even if it explicitly requires them setting an evar to do so.

Reply:
with the _DEBUG as part of the name - the original name is better.